lib/socket/handlers.js

```diff
@@ -520,6 +523,46 @@ module.exports =  function(socketUtils){
     }, socket.request.sessionId, 'ctrl');
   });
 
+  var updatePresentationSettings = coroutine(function *updatePresentationSettingsGen(socket, evt) {
+    console.log('updatePresentationSettings');
+    var user = socket.decoded_token.user;
+    var userId = user ? user._id : null;
+    var presentation = yield Slideshow.findById(evt.presentationId).exec();
+
+    if ( !userId ) {
+      socket.emit('asq:update_presentation_settings_ack', {'state': false});
```

Comment:
I am a little bit confused about the ack event. ack means acknowledgement, as in 'I received what you sent me'. You are using it more as a response for the result of the request rather than acknowledgement.